src/server/scripts/Kalimdor/HallsOfOrigination/boss_isiset.cpp

```diff
@@ -0,0 +1,1071 @@
+    SPELL_MANA_SHIELD_STATE_VISUAL = 74266,
+    SPELL_IMAGE_EXPLOSION = 74301,
+
+    // Celestial Familiar npc
+    SPELL_ORB_VISUAL = 74356, // Adds State Visual (orb)
+    SPELL_ARCANE_BARRAGE = 74374,
+
+    // Astral Shift Explosion Visual npc
+    SPELL_ASTRAL_SHIFT_EXPLOSION_VISUAL = 74331,
+
+    // Starry Sky npc
+    SPELL_STARRY_SKY_VISUAL = 74149,
+};
+
+enum NPCs
+{
+    NPC_CELESTIAL_FAMILIAR = 39795,
+    NPC_ASTRAL_RAIN = 39720, // N
+    NPC_CELESTIAL_CALL = 39721, // E
+    NPC_VEIL_OF_SKY = 39722, // W
+};
+
+enum Texts
+{
+    SAY_AGGRO = 0,
+    SAY_SUPERNOVA = 1,
+    SAY_SUPERNOVA_WARNING = 2,
+    SAY_PLAYER_DEATH = 3,
+    SAY_DEATH = 4,
+};
+
+enum Events
+{
+    EVENT_NONE,
+    EVENT_START_ATTACK,
+    EVENT_SUPERNOVA,
+    EVENT_ASTRAL_RAIN,
+    EVENT_ASTRAL_FAMILIAR,
+    EVENT_VEIL_OF_SKY,
+    EVENT_IMAGES_ATTACK,
+    EVENT_DESPAWN_IMAGES,
+
+    EVENT_ORB_ADD_VISUAL,
+    EVENT_ORB_SET_AGGRESSIVE,
+    EVENT_ORB_ARCANE_BARRAGE,
+
+    EVENT_IMAGE_ASTRAL_RAIN,
+    EVENT_IMAGE_ARCANE_BARRAGE,
+    EVENT_IMAGE_VEIL_OF_SKY,
+
+    EVENT_STARRY_SKY_ADD_VISUAL,
+
+    EVENT_ENERGY_FLUX,
+};
+
+enum Actions
+{
+    ACTION_NONE,
+    ACTION_IMAGES_SET_AGGRESSIVE,
+    ACTION_MIRROR_IMAGE_DIED,
+    ACTION_IMAGES_SET_PASSIVE,
+    ACTION_IMAGES_DESPAWN,
+};
+
+class boss_isiset : public CreatureScript
+{
+public:
+    boss_isiset() : CreatureScript("boss_isiset") { }
+
+    struct boss_isisetAI : public BossAI
+    {
+        boss_isisetAI(Creature* creature) : BossAI(creature, DATA_ISISET) { }
+
+        void Reset() override
+        {
+            _Reset();
+
+            me->SetReactState(REACT_AGGRESSIVE);
+
+            instance->SetData(DATA_ISISET_PHASE, 1);
+            instance->SetData(DATA_ISISET_ASTRAL_RAIN_ALIVE, 1);
+            instance->SetData(DATA_ISISET_CELESTIAL_CALL_ALIVE, 1);
+            instance->SetData(DATA_ISISET_VEIL_OF_SKY_ALIVE, 1);
+
+            RescheduleEvents();
+        }
+
+        void EnterCombat(Unit* /*victim*/) override
+        {
+            instance->SendEncounterUnit(ENCOUNTER_FRAME_SET_COMBAT_RES_LIMIT, 0);
+            me->SetReactState(REACT_PASSIVE);
+            Talk(SAY_AGGRO);
+            _EnterCombat();
+        }
+
+        void DamageTaken(Unit* /*attacker*/, uint32 &damage) override
+        {
+            if (me->HasAura(SPELL_ASTRAL_SHIFT))
+                return;
+
+            // Second transition
+            if (instance->GetData(DATA_ISISET_PHASE) == 2 && me->HealthBelowPctDamaged(33, damage))
+                MirrorImage();
+
+            // First transition
+            if (instance->GetData(DATA_ISISET_PHASE) == 1 && me->HealthBelowPctDamaged(66, damage))
+                MirrorImage();
+        }
+
+        void JustSummoned(Creature* summoned) override
+        {
+            switch (summoned->GetEntry())
+            {
+            case NPC_ASTRAL_RAIN:
+                instance->SendEncounterUnit(ENCOUNTER_FRAME_ENGAGE, summoned, 1);
```

Comment:
this creatures has core scripts? it can be handled on Reset() or EnterCombat() of this creatures no?